ERCS/erc-7638.md

```diff
@@ -0,0 +1,195 @@
+
+In addition to the atomic operation of `approve-transferFrom` mentioned above, gas payment delegation can also be achieved. It involves users and bundlers signing a set of calls, where the content of the calls includes:
+
+1. The user wishes to initiate multiple calls through his SCA.
+2. The user transfers 10 USDT to the bundler as fee, included within the calls.
+3. The bundler submits the calls, pay ETH gas and get the 10 USDT.
+
+The user encodes the content of the calls, attaches their signature to ensure its integrity, and sends it to the bundler. If the bundler considers the gas payment insufficient, they may choose not to submit it. However, if they approve the content of the calls, the signed transaction can be submitted. After execution, the user obtains the desired operations, and the bundler receives the fee.
+
+[EIP-4337](./eip-4337.md) also implements gas payment delegation. BCE and [EIP-4337](./eip-4337.md) are not mutually exclusive and can be implemented concurrently within an SCA.
+
+Based on empirical testing, BCE is simpler and more gas-efficient compared to alternative methods.
+
+## Specification
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+This ERC **REQUIRED** SCA to be implemented in the contract, where the Dapp communicates with the SCA wallet extension to communicate the user's intentions to the wallet, which uses Batch Calls Encoding to send multiple calls as bytes to the user's SCA contract.
+
+`Batch Calls` comprises multiple `Call` bytes, each defined by the encoding of `to\value\data` as follows:
+
+```mermaid
+graph LR
+A["to (20bytes)"] --- B["value (32bytes)"] --- C["data length (32bytes)"] --- D["data (bytes)"]
+```
+
+Let:
+- `to`: The address of the called contract, corresponding to the Solidity address type, 20 bytes.
+- `value`: The amount of ETH(in wei) sent to the contract, in wei, corresponding to the Solidity uint type, 32 bytes.
+- `data length`: The length of the data(in bytes), corresponding to the Solidity uint type, 32 bytes.
+- `data`: The encoded functionData sent to the contract, corresponding to the Solidity bytes type, with a length defined by `data length`.
+
+Multiple `Call` units are concatenated to form an `Batch Calls` sequence.
+
+
+## Rationale
+Each call encapsulates 3 parameters: `to\vallue\data`. The conventional approach involves packaging these 3 parameters into a struct and then placing multiple structs into an array. However, using a struct adds overhead as it also packages the types of `to\vallue\data`, increasing the size of the encoding. Since `to\vallue\data` have fixed types, this additional encoding can be omitted. In Solidity, reading data from `bytes calldata` using slice is a gas-efficient method. Considering these factors, `Batch Calls Encoding` can compress on-chain data and save gas.
```

Comment:
```suggestion
Each call encapsulates 3 parameters: `to`\`value`\`data`. The conventional approach involves packaging these 3 parameters into a struct and then placing multiple structs into an array. However, using a struct adds overhead as it also packages the types of `to`\`value`\`data`, increasing the size of the encoding. Since `to`\`value`\`data` have fixed types, this additional encoding can be omitted. In Solidity, reading data from `bytes calldata` using slice is a gas-efficient method. Considering these factors, _Batch Calls Encoding_ can compress on-chain data and save gas.
```